Sources/SwiftDocCUtilities/Action/Actions/Convert/ConvertAction.swift

```diff
@@ -370,6 +371,11 @@ public struct ConvertAction: Action, RecreatingContext {
         
         // If we're building a navigation index, finalize the process and collect encountered problems.
         if let indexer = indexer {
+            let finalizeNavigationIndexMetric = benchmark(begin: Benchmark.Duration(id: "finalize-navigation-index"))
+            defer {
```

Comment:
Does `defer` work for _any_ scope? I thought it always caused execution at the end of the function but that doesn't seem to be the case based on this and the benchmarking results.

Reply:
Yes, if you run something like this:
```swift
print("Start")
defer { print("End") }
for i in 0...5 {
    defer { print("Loop - end \(i)")}
    print("Loop - start \(i)")
    
    if i.isMultiple(of: 2) {
        defer { print("If inside loop - end \(i)")}
        print("If inside loop - start \(i)")
    }
}
if !"something".isEmpty {
    defer { print("If - end")}
    print("If - start")
}
```
it will print
```
Start
Loop - start 0
If inside loop - start 0
If inside loop - end 0
Loop - end 0
Loop - start 1
Loop - end 1
Loop - start 2
If inside loop - start 2
If inside loop - end 2
Loop - end 2
Loop - start 3
Loop - end 3
Loop - start 4
If inside loop - start 4
If inside loop - end 4
Loop - end 4
Loop - start 5
Loop - end 5
If - start
If - end
End
```

Reply:
Oh wild. I had no idea 😃 Thanks!